app/models/job_queue.rb

```diff
@@ -104,7 +110,13 @@ def perform_job(job_execution, queue)
 
     @threads[job_execution.id] = Thread.new do
       begin
-        job_execution.perform
+        ActiveRecord::Base.connection_pool.with_connection do
+          begin
+            job_execution.perform
+          rescue Cancel
```

Comment:
what about successful deploys, don't want to use `ensure` and throw it away too?